msbuild/Xamarin.MacDev.Tasks.Core/Tasks/CompileEntitlementsTaskBase.cs

```diff
@@ -304,15 +305,35 @@ protected virtual MobileProvision GetMobileProvision (MobileProvisionPlatform pl
 
 		public override bool Execute ()
 		{
+			MobileProvisionPlatform platform;
 			MobileProvision profile;
 			PDictionary template;
 			PDictionary compiled;
 			PDictionary archived;
 			string path;
 			bool save;
 
+			switch (SdkPlatform) {
+			case "AppleTVSimulator":
+			case "AppleTVOS":
+				platform = MobileProvisionPlatform.tvOS;
+				break;
+			case "iPhoneSimulator":
+			case "WatchSimulator":
+			case "iPhoneOS":
+			case "WatchOS":
+				platform = MobileProvisionPlatform.iOS;
```

Comment:
Shouldn't you have a `watchOS` case as well?

Reply:
It already has `case "WatchOS":`

Reply:
Sorry, I meant to point out that the `WatchOS` case points to `MobileProvisionPlatform.iOS`. Is that intended?

Reply:
Yes. I don't think there is a watchOS-specific platform value in provisioning profiles.

FWIW, the old code was hard-coded to iOS.

Reply:
👍

Reply:
It's not clear to me either. Perhaps we just got lucky that the provisioning profiles we had targeted both iOS and tvOS? Or maybe it never worked and we've just been ignoring the warnings this whole time?